extensions/kafka/src/main/java/io/deephaven/kafka/KafkaTools.java

```diff
@@ -575,19 +576,50 @@ DataFormat dataFormat() {
              */
             static final class Json extends KeyOrValueSpec {
                 final ColumnDefinition<?>[] columnDefinitions;
-                final Map<String, String> fieldNameToColumnName;
+                final Map<String, String> fieldToColumnName;
 
                 private Json(
                         final ColumnDefinition<?>[] columnDefinitions,
                         final Map<String, String> fieldNameToColumnName) {
                     this.columnDefinitions = columnDefinitions;
-                    this.fieldNameToColumnName = fieldNameToColumnName;
+                    this.fieldToColumnName = mapNonPointers(fieldNameToColumnName);
                 }
 
                 @Override
                 DataFormat dataFormat() {
                     return DataFormat.JSON;
                 }
+
+                private static Map<String, String> mapNonPointers(final Map<String, String> fieldNameToColumnName) {
+                    if (fieldNameToColumnName == null) {
+                        return null;
+                    }
+                    boolean needsMapping = false;
```

Comment:
This actually seems like a case streams would be clearer:
final boolean needsMapping = fieldNameToColumnName.keySet().anymatch(key -> key.startsWith("/"));

This is not necessary, but we have very similar loops to parse out.

Reply:
Done.